File: Lecture033 Recursion Day3/sortedArray.py

def isSorted(arr, size):

    # base case
    if(size == 0 or size == 1):
        return True
    

    if(arr[0] > arr[1]):
        return False
    else:
        rem_arr = arr[1:]
        remainingPart = int(isSorted(rem_arr, len(rem_arr)))
        return remainingPart
    



if __name__ == '__main__':

    arr =[2, 3, 9, 9, 9]
    size = 5

    ans = int(isSorted(arr, size))

    if(ans):
        print("Array is sorted ")
    
    else:
        print("Array is not sorted ")
    

File: Lecture033 Recursion Day3/sortedArray.cpp
#include<iostream>
using namespace std;

bool isSorted(int arr[], int size) {

    //base case
    if(size == 0 || size == 1 ){
        return true;
    }

    if(arr[0] > arr[1])
        return false;
    else {
        bool remainingPart = isSorted(arr + 1, size - 1 );
        return remainingPart;
    }

}

int main() {

    int arr[5] = {2,4,9,9,9};
    int size = 5;

    bool ans = isSorted(arr, size);

    if(ans){
        cout << "Array is sorted " << endl;
    }
    else {
        cout << "Array is not sorted " << endl;
    }

    return 0;
}